libs/isograph-react/src/core/makeNetworkRequest.ts

```diff
@@ -44,6 +47,14 @@ export function maybeMakeNetworkRequest<
       return [wrapResolvedValue(undefined), () => {}];
     }
     case 'IfNecessary': {
+      if (
+        artifact.networkRequestInfo.normalizationAst.kind ===
+        'NormalizationAstLoader'
+      ) {
+        throw new Error(
+          'Using lazy loaded normalizationAst with shouldFetch: "IfNecessary" is not supported as it will lead to slower initial load time.',
```

Comment:
I'm not sold on this comment, tbh, but let's figure out what to say here in the future